cpp_utils/src/cpp/qos/qos_utils.cpp

```diff
@@ -0,0 +1,143 @@
+            return "Writer data lifecycle";
+        case QosPolicyId_t::READERDATALIFECYCLE_QOS_POLICY_ID:
+            return "Reader data lifecycle";
+        case QosPolicyId_t::TOPICDATA_QOS_POLICY_ID:
+            return "Topic data";
+        case QosPolicyId_t::GROUPDATA_QOS_POLICY_ID:
+            return "Group data";
+        case QosPolicyId_t::TRANSPORTPRIORITY_QOS_POLICY_ID:
+            return "Transport priority";
+        case QosPolicyId_t::LIFESPAN_QOS_POLICY_ID:
+            return "Lifespan";
+        case QosPolicyId_t::DURABILITYSERVICE_QOS_POLICY_ID:
+            return "Durability service";
+        case QosPolicyId_t::DATAREPRESENTATION_QOS_POLICY_ID:
+            return "Data representation";
+        case QosPolicyId_t::TYPECONSISTENCYENFORCEMENT_QOS_POLICY_ID:
+            return "Type consistency enforcement";
+        case QosPolicyId_t::DISABLEPOSITIVEACKS_QOS_POLICY_ID:
+            return "Disable positive acks";
+        case QosPolicyId_t::PARTICIPANTRESOURCELIMITS_QOS_POLICY_ID:
+            return "Participant resource limits";
+        case QosPolicyId_t::PROPERTYPOLICY_QOS_POLICY_ID:
+            return "Property policy";
+        case QosPolicyId_t::PUBLISHMODE_QOS_POLICY_ID:
+            return "Publish mode";
+        case QosPolicyId_t::READERRESOURCELIMITS_QOS_POLICY_ID:
+            return "Reader resource limits";
+        case QosPolicyId_t::RTPSENDPOINT_QOS_POLICY_ID:
+            return "RTPS endpoint";
+        case QosPolicyId_t::RTPSRELIABLEREADER_QOS_POLICY_ID:
+            return "RTPS reliable reader";
+        case QosPolicyId_t::RTPSRELIABLEWRITER_QOS_POLICY_ID:
+            return "RTPS reliable writer";
+        case QosPolicyId_t::TRANSPORTCONFIG_QOS_POLICY_ID:
+            return "Transport config";
+        case QosPolicyId_t::TYPECONSISTENCY_QOS_POLICY_ID:
+            return "Type consistency";
+        case QosPolicyId_t::WIREPROTOCOLCONFIG_QOS_POLICY_ID:
+            return "Wire protocol config";
+        case QosPolicyId_t::WRITERRESOURCELIMITS_QOS_POLICY_ID:
+            return "Writer resource limits";
+        case QosPolicyId_t::INVALID_QOS_POLICY_ID:
+        default:
+            return "Invalid";
+    }
+}
+
+std::string qos_policy_mask_to_string(
+        const PolicyMask& qos_policy_mask)
+{
+    if (qos_policy_mask.none())
+    {
+        return qos_policy_id_to_string(QosPolicyId_t::INVALID_QOS_POLICY_ID);
+    }
+
+    std::string policy_mask_str;
+
+    for (std::uint32_t id_int = QosPolicyId_t::INVALID_QOS_POLICY_ID; id_int < QosPolicyId_t::NEXT_QOS_POLICY_ID;
+            id_int++)
+    {
+        const auto qos_policy_id = static_cast<QosPolicyId_t>(id_int);
+
+        if (qos_policy_mask.test(qos_policy_id))
+        {
+            policy_mask_str += ", " + qos_policy_id_to_string(qos_policy_id);
+        }
+    }
+
+    if (!policy_mask_str.empty())
+    {
+        policy_mask_str.erase(0, 2);
+    }
```

Comment:
I prefer not to delete the initial ' ,' from the string. While there's no perfect solution, one approach is to add a parameter to track whether the `QosPolicy` is the first in the sequence.